Fix link in the top header.

Nielius opened the pull request:
When I capture something, the header is empty instead of containing a link to the website. This fixes it.

alphapapa commented:
Hi,

Thanks for sending this PR, but I'm not sure exactly what the problem is or how this fixes it.  Can you show me an example of what happens without this fix?  Also, can you tell me what version of Org you're using, and how you're running the capture (i.e. are you using Firefox with the protocol handler, or the shell script, or...)?  Since I haven't noticed the problem, I wonder if we're using different versions of something; if so, I'd like to document it.  :)

Thanks.

Nielius commented:
Sorry for the late reply. I was away for some time (and will be away again in the next few weeks, so don't expect anything from me anytime soon ;) ).

Maybe the best way to describe the problem is  by giving an example: when I try to capture the site [http://kitchingroup.cheme.cmu.edu/blog/2014/07/17/Pandoc-does-org-mode-now/], I get:
```
** 
  
  ** [[/blog/2014/07/17/Pandoc-does-org-mode-now/][Pandoc does org-mode now]]
 
  Posted July 17, 2014 at 10:04 AM | categories: [[/blog/category/uncategorized/][uncategorized]] | tags: | [[http://jkitchin.github.io/blog/2014/07/17/Pandoc-does-org-mode-now#disqus_thread][6 Comments]]
 
  ** Table of Contents
 
  -  [[#sec-1][1. A subsection with some equations]]
  -  [[#sec-2][2. A section with a figure]]
....
```

So as you see, the top header is just empty (it's only "**") and there is no link to the website anywhere. I fixed it by adding the one line in the pull request.

I'm not sure how it works, but from what I understand, the capture template uses the information that is stored by the function org-store-link-props. For some reason, if my templates used "%l" (AFAIK supposed to refer to the link) or "%a", I would get nothing, but by adding this additional property, it worked.

So the problem could also be the capture templates. I tried the following templates, but neither worked:
```
(setq org-capture-templates
      '(("l" "Link" entry (file+olp "~/Dropbox/org/notes.org" "Web Links")
	 "* %a\n %?\n %i")
	("w" "Web link (test)" entry (file+olp "~/Dropbox/org/notes.org" "Web Links")
	"* %l \n %?\n %i")))
```

What does your template look like? You could also consider putting your template in the readme - that would have been helpful for me, too. (I haven't really used these templates for anything else.)

And concerning the software I'm using: I'm using Kubuntu 14.04, just updated everything today, and (emacs-version) and (org-version) give the following:
- GNU Emacs 25.1.50.2 (x86_64-pc-linux-gnu, GTK+ Version 3.10.8) of 2016-02-29
- Org-mode version 8.3.4 (8.3.4-5-gdc68d2-elpa @ /home/niels/.emacs.d/elpa/org-20160229/)

alphapapa commented:
Thanks.  I think the issue is probably that you are using a newer Emacs and/or Org-mode than I am.  I'm also on Kubuntu 14.04, but I haven't installed current Emacs and Org like you have.  I'll check the Org changelog; I suspect that something in Org 8.3 changed the templates or capturing.  I'll figure it out and either make a fix or add a note about the templates.  Have a good trip.  :)

Nielius commented:
Ok, thanks.

alphapapa commented:
I just made some changes, including setting the `:annotation` of the `org-link`, and adding a recommended `org-capture` template.  I think if you use the new code and the template from the readme, it will probably work for you.  Please let me know what you find out.  :)

Nielius commented:
Yes, it works for me (with either my own or your template). Thanks!

alphapapa commented:
Thanks!  By the way, if you have time, I would appreciate any feedback on the new `org-protocol-capture-readability` function.  It's working well for me, but I'd like to hear what it's like for others.  :)
